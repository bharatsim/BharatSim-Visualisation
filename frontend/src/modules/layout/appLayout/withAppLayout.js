import React from 'react';
import Header from './header/Header';

function withAppLayout(WrappedComponent) {
  return function ComponentWithHeader() {
    return (
      <div>
        <Header />
        <WrappedComponent />
      </div>
    );
  };
}

export default withAppLayout;
