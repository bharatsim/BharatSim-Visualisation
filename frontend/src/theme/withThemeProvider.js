import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
