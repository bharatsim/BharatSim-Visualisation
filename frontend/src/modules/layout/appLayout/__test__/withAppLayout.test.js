import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import withAppLayout from '../withAppLayout';
import withThemeProvider from '../../../../theme/withThemeProvider';

function DummyController() {
  return <div>DummyController</div>;
}

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const Component = withThemeProvider(withAppLayout(DummyController));

describe('<Header />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });

  it('should get to landing on click of logo', () => {
    const { getByAltText } = render(<Component />);
    fireEvent.click(getByAltText('logo'));

    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });
});
