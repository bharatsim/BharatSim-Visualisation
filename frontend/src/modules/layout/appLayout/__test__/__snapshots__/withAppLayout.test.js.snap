// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <div>
    <header
      class="MuiBox-root MuiBox-root-3 makeStyles-mainContainer-1"
      cursor="pointer"
    >
      <div
        class="MuiBox-root MuiBox-root-4 makeStyles-logo-2"
      >
        <img
          alt="logo"
          src="logo.svg"
          width="200"
        />
      </div>
    </header>
    <div>
      DummyController
    </div>
  </div>
</div>
`;
